ingestion/src/metadata/data_quality/processor/test_case_runner.py

```diff
@@ -0,0 +1,307 @@
+#  Copyright 2021 Collate
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""
+This Processor is in charge of executing the test cases
+"""
+import traceback
+from copy import deepcopy
+from typing import List, Optional, cast
+
+from metadata.data_quality.api.models import (
+    TableAndTests,
+    TestCaseDefinition,
+    TestCaseResultResponse,
+    TestCaseResults,
+    TestSuiteProcessorConfig,
+)
+from metadata.data_quality.runner.core import DataTestsRunner
+from metadata.data_quality.runner.test_suite_source_factory import (
+    test_suite_source_factory,
+)
+from metadata.generated.schema.api.tests.createTestCase import CreateTestCaseRequest
+from metadata.generated.schema.metadataIngestion.workflow import (
+    OpenMetadataWorkflowConfig,
+)
+from metadata.generated.schema.tests.testCase import TestCase
+from metadata.generated.schema.tests.testDefinition import TestDefinition, TestPlatform
+from metadata.generated.schema.tests.testSuite import TestSuite
+from metadata.generated.schema.type.basic import EntityLink, FullyQualifiedEntityName
+from metadata.ingestion.api.models import Either, StackTraceError
+from metadata.ingestion.api.parser import parse_workflow_config_gracefully
+from metadata.ingestion.api.step import Step
+from metadata.ingestion.api.steps import Processor
+from metadata.ingestion.ometa.ometa_api import OpenMetadata
+from metadata.utils import entity_link, fqn
+from metadata.utils.logger import test_suite_logger
+
+logger = test_suite_logger()
+
+
+class TestCaseRunner(Processor):
+    """Execute the test suite tests and create test cases from the YAML config"""
+
+    def __init__(self, config: OpenMetadataWorkflowConfig, metadata: OpenMetadata):
+
+        super().__init__()
+
+        self.config = config
+        self.metadata = metadata
+
+        self.processor_config: TestSuiteProcessorConfig = (
+            TestSuiteProcessorConfig.parse_obj(
+                self.config.processor.dict().get("config")
+            )
+        )
+
+    def _run(self, record: TableAndTests) -> Either:
+        # First, create the executable test suite if it does not exist yet
+        # This could happen if the process is executed from YAML and not the UI
+        if record.executable_test_suite:
+            # We pass the test suite request to the sink
+            return Either(right=record.executable_test_suite)
```

Comment:
Wouldn't this be handle in `source/test_suite.py` ln 172? If we return this in this step wouldn't we be just passing the testSuite to the sink?

Reply:
yes exactly that. In the source we return 2 possibilities:
1. First, if needed, the executable test suite to be created. In that case, the processor does not need to do anything, just pass it to the sink to get it created
2. Then, pick up the test cases (if any) and return them with the table. Now it's where the processor needs to get the cases from the CLI, execute them etc.

Reply:
so in the case where we have a table without executable test suite, the source will first return the suite to be created, and then the rest of the logic will happen